foursight_core/react/api/react_api_base.py

```diff
@@ -179,7 +179,34 @@ def react_authorize(self, request: dict, env: Optional[str]) -> dict:
         """
         return self._auth.authorize(request, env)
 
+    def get_this_base_url(self, request: dict) -> str:
+        """
+        Returns the base URL of this Foursight instance.
+        """
+        domain, context = app.core.get_domain_and_context(request)
+        if is_running_locally(request):
+            scheme = "http"
+            context = ROUTE_PREFIX
+        else:
+            scheme = "https"
+        if context.endswith("/"):
+            context = context[0:-1]
+        return f"{scheme}://{domain}{context}"
+
     def get_redirect_url(self, request: dict, env: str, domain: str, context: str) -> str:
+        """
+        Returns the redirect URL to the UI from the reactredir cookie, or if that
```

Comment:
The frontend (React) stores a cooke ("reactredir") which is the URL of the most recently visited page, so that on logout and subsequent re-login the user will be take back to that page (this kind of thing was done in the previous non-React Foursight). So this function just gets that URL cookie and returns it, or returns the URL to the /login page if not set. Not sure that answers your question;

Reply:
Yes, it would, if you mean cookies from one tab carrying over to the other; depending on the situation a blessing or curse; if you know a way to control this though I'd be interested.